src/deepl.ts

```diff
@@ -22,16 +22,52 @@ export class DeepLApi {
       url: "/translate",
       data: qs.stringify({
         auth_key: this.authKey,
-        text: text,
-        target_lang: targetLanguage.toUpperCase()
+        text: text.replace(/<(.*?)>/g, function(i: any, match: string) {
+          if(match.match(/^!subteam/)) {
+            return '';
+          }
+          if(match.match(/^!/)) {
+            const matched = match.match(/^!([a-z]+)$/);
+            if (matched != null) {
+              return '<ignore>@' + matched[1] + '</ignore>';
+            }
+            return '';
+          }
+          if(match.match(/^.*?\|.*$/)) {
+            const matched = match.match(/^(.*?)\|(.*)$/);
+            if (matched != null) {
+              return '<a href="' + matched[1] + '">' + matched[2] + '</a>';
+            }
+            return '';
+          } else {
+            return '<url>' + match + '</url>';
```

Comment:
This means the `match` is not necessarily a URL, right? If so, this workaround makes sense to me. 

One possibility that I haven't verified is... What happens if a user writes a message like "I think < and > should be kept as-is"? Perhaps, your current code should work for the case too. Am I right?

Reply:
Slack encodes user-written `< >` as `&lt; &gt;`, so there is no collision. I just tested it, it works fine.
See also the overall cleanup and documentation I have done. Details in the new comment.